sql/core/src/main/scala/org/apache/spark/sql/json/JsonRDD.scala

```diff
@@ -127,7 +135,7 @@ private[sql] object JsonRDD extends Logging {
       StructType((topLevelFields ++ structFields).sortBy(_.name))
     }
 
-    makeStruct(resolved.keySet.toSeq, Nil)
+    nullTypeToStringType(makeStruct(resolved.keySet.toSeq, Nil))
```

Comment:
Yeah, I understand that, but this is in the `createSchema` method, it supposes the `union` operations have been done before this method is called, doesn't it?
see https://github.com/chenghao-intel/spark/blob/json_inferschema/sql/core/src/main/scala/org/apache/spark/sql/json/JsonRDD.scala#L53
https://github.com/chenghao-intel/spark/blob/json_inferschema/sql/core/src/main/scala/org/apache/spark/sql/json/JsonRDD.scala#L61

Sorry, I am not sure if you are talking about to merge 2 `StructType` in the future, if it is, I will update the code.